source/grid/grid_generator.cc

```diff
@@ -5360,25 +5361,96 @@ namespace GridGenerator
 
     std::map<typename MeshType<dim - 1, spacedim>::cell_iterator,
              typename MeshType<dim, spacedim>::face_iterator>
-      surface_to_volume_mapping;
+      surface_to_volume_mapping; // return of the function
 
     const unsigned int boundary_dim = dim - 1; // dimension of the boundary mesh
 
-    // First create surface mesh and mapping
-    // from only level(0) cells of volume_mesh
-    std::vector<typename MeshType<dim, spacedim>::face_iterator>
+    std::vector<
+      std::pair<typename MeshType<dim, spacedim>::face_iterator, unsigned int>>
       mapping; // temporary map for level==0
+               // iterator to face is stored along with face number
+               //(this is required by the algorithm to adjust the normals of the
+               // cells of the boundary mesh)
 
+    std::vector<bool> touched(
+      volume_mesh.get_triangulation().n_vertices(),
+      false); // vector indicating whether a vertex of the volume mesh has
+              // already been visited (necessary to avoid duplicate vertices in
```

Comment:
We typically put comments in front of expressions. That way they are formatted to use the whole width.

Reply:
Ok, I often put comments to the side when they refer only to a single line and in front if referring to a whole bunch of comments. But of course, I'll adapt to the dealii style here.